### PR TITLE
[SDESK-4251] Refactor the article side-tab widgets and consolidate th…

### PR DESCRIPTION
**Pull request by fritzSF**
…e list views in them

### CODE REVIEW

**Comment by pavlovicnemanja** [comment]
Depends on merging #2957

File: scripts/apps/authoring/widgets/widgets-article.scss [hunk 1]
```diff
@@ -39,9 +39,10 @@
 
 .tabpane {
     .sd-widget {
-        width: auto; height: auto;
-        position: absolute;
-        top: 0; left: 0; right: 0; bottom: 0;
+        width: auto; 
+        height: 100%;
+        position: relative;
+        //top: 0; left: 0; right: 0; bottom: 0;
```
[the remark this answers is not in the document]
Comment (fritzSF) [reply scripts/apps/authoring/widgets/widgets-article.scss]: Most of the translate filters are not added, they were already there. Also, translate attribute creates an additional span around the element, which is not desired in some cases. We can't get rid of those in all cases anyway, we need it for titles, input placeholders, toggle boxes etc.

**Comment by fritzSF** [comment]
> `python3 grep-lint.py` fails due to addition of translate filters. Replace `translate` filters with attributes. `grep-lint` script will always run on build when #2950 is merged.

Most of the translate filters are not added, they were already there. Also, translate attribute creates an additional span around the element, which is not desired in some cases. We can't get rid of those in all cases anyway, we need it for titles, input placeholders, toggle boxes etc.

**Comment by tomaskikutis** [comment]
The script only complains if filter usages count is greater than on master. Additional span is created with the usage of `translate` attribute, but it won't cause issues most of the time.

Regarding the translations of usual attributes like `title`, we should be using a `gettext` function. `<button title="gettext('Remove)">x</button>`. The same one should work inside an element without creating an additional span: `<button>{{gettext('Remove')}}<button>`. The downside is that you might need to add `gettext` to the scope. I'm not sure angular allows using global variables in templates.

**Comment by MarkLark86** [comment]
> Regarding the translations of usual attributes like `title`, we should be using a `gettext` function. 

@tomaskikutis What is the problem with using angular-gettext/filters?

**Comment by tomaskikutis** [comment]
@MarkLark86 strings aren't getting extracted for translation by `nggettext_extract` script if you use filters.

**Pull request merged**